File: cgi-bin/py/medium.py
#!/usr/bin/env python3
import sys
import time
import os

# Get a unique ID from query string
query = os.environ.get('QUERY_STRING', '')
cgi_id = query.split('=')[1] if '=' in query else 'unknown'

# Send headers immediately
sys.stdout.write("Content-Type: text/html\r\n")
sys.stdout.write("\r\n")
sys.stdout.flush()

# Sleep 1.5 seconds to simulate medium processing
time.sleep(1.5)

# Send response
print(f"<html><body>")
print(f"<h1>CGI #{cgi_id} completed!</h1>")
print(f"<p>This CGI took 1.5 seconds to process.</p>")
print(f"</body></html>")

File: cgi-bin/php/medium.php
#!/usr/bin/env php-cgi
<?php
// Get a unique ID from query string
$query = isset($_SERVER['QUERY_STRING']) ? $_SERVER['QUERY_STRING'] : '';
$cgi_id = 'unknown';
if (strpos($query, '=') !== false) {
    $parts = explode('=', $query, 2);
    $cgi_id = $parts[1];
}

// Send headers immediately
header('Content-type: text/html');

// Flush output before sleeping
ob_implicit_flush(true);
ob_end_flush();

// Sleep 1.5 seconds to simulate medium processing
usleep(1500000);

// Send response
echo "<html><body>\n";
echo "<h1>CGI #$cgi_id completed!</h1>\n";
echo "<p>This CGI took 1.5 seconds to process.</p>\n";
echo "</body></html>\n";
?>
